core/file_walker.go

```diff
@@ -40,17 +37,16 @@ func (f *File) UpdateSize() {
 
 type ReadDir func(dirname string) ([]os.FileInfo, error)
 
-func GetSubTree(path string, parent *File, readDir ReadDir, ignoredFolders map[string]struct{}) *File {
-	var mutex sync.Mutex
+func WalkFolder(path string, readDir ReadDir, ignoredFolders map[string]struct{}) *File {
 	var wg sync.WaitGroup
-	c := make(chan bool, maxConcurrentScans)
-	root := getSubTreeConcurrently(path, parent, readDir, ignoredFolders, c, &mutex, &wg)
+	c := make(chan bool, runtime.NumCPU())
```

Comment:
I don't know if that's a good idea. Please do some benchmarks first. In my tests, 16 concurrent scans were faster than 32 (however I only have 8 cores).